sdk/storage/storage-blob/samples/javascript/iterators-without-await.js

```diff
@@ -0,0 +1,62 @@
+/*
+ Setup: Enter your storage account name and shared key in main()
+*/
+
+const { BlobServiceClient, StorageSharedKeyCredential } = require("../../"); // Change to "@azure/storage-blob" in your package
+
+async function main() {
+  // Enter your storage account name and shared key
+  const account = process.env.ACCOUNT_NAME || "";
+  const accountKey = process.env.ACCOUNT_KEY || "";
+
+  // Use StorageSharedKeyCredential with storage account and account key
+  // StorageSharedKeyCredential is only avaiable in Node.js runtime, not in browsers
+  const sharedKeyCredential = new StorageSharedKeyCredential(account, accountKey);
+
+  const blobServiceClient = new BlobServiceClient(
+    `https://${account}.blob.core.windows.net`,
+    sharedKeyCredential
+  );
+
+  // Create a container
+  const containerName = `newcontainer${new Date().getTime()}`;
+  const containerClient = blobServiceClient.getContainerClient(containerName);
+
+  const createContainerResponse = await containerClient.create();
+  console.log(`Created container ${containerName} successfully`, createContainerResponse.requestId);
+
+  for (let index = 0; index < 7; index++) {
+    // Create a blob
+    const content = "hello";
+    const blobName = "newblob" + new Date().getTime();
+    const blobClient = containerClient.getBlobClient(blobName);
+    const blockBlobClient = blobClient.getBlockBlobClient();
+    const uploadBlobResponse = await blockBlobClient.upload(content, content.length);
+    console.log(`Uploaded block blob ${blobName} successfully`, uploadBlobResponse.requestId);
+  }
+
+  console.log("Listing all blobs without await");
+  let i = 1;
+  let asyncIter = containerClient.listBlobsFlat();
+
+  function printBlob(result) {
+    if (!result.done) {
+      console.log("Blob " + (i++) + ": " + result.value.name);
+      asyncIter.next().then(printBlob);
+    }
+  }
+
+  asyncIter.next().then(printBlob);
+
+  await containerClient.delete();
+  console.log("deleted container");
+}
+
+// An async method returns a Promise object, which is compatible with then().catch() coding style.
+main()
```

Comment:
There is another ongoing PR about updating samples execution which removed main entry execution.